src/webgl/primitives.js

```diff
@@ -769,66 +721,65 @@ p5.RendererGL.prototype.ellipse = function(args) {
   //detailX and Y are optional 6th & 7th
   //arguments
   var detailX = args[4] || 24;
-  var detailY = args[5] || 16;
-  var gId =
-    'ellipse|' + args[0] + '|' + args[1] + '|' + args[2] + '|' + args[3];
+  var gId = 'ellipse|' + detailX;
   if (!this.geometryInHash(gId)) {
     var _ellipse = function() {
-      var u, v, p;
-      var centerX = x + width * 0.5;
-      var centerY = y + height * 0.5;
-      for (var i = 0; i <= this.detailY; i++) {
-        v = i / this.detailY;
-        for (var j = 0; j <= this.detailX; j++) {
-          u = j / this.detailX;
-          var theta = 2 * Math.PI * u;
-          if (v === 0) {
-            p = new p5.Vector(centerX, centerY, 0);
-          } else {
-            var _x = centerX + width * 0.5 * Math.cos(theta);
-            var _y = centerY + height * 0.5 * Math.sin(theta);
-            p = new p5.Vector(_x, _y, 0);
-          }
-          this.vertices.push(p);
-          this.uvs.push([u, v]);
-        }
+      this.vertices.push(new p5.Vector(0, 0, 0));
+      this.uvs.push([0, 0]);
+
+      for (var i = 0; i <= this.detailX; i++) {
+        var u = i / this.detailX;
+        var theta = 2 * Math.PI * u;
+
+        var _x = Math.cos(theta) / 2;
+        var _y = Math.sin(theta) / 2;
+
+        this.vertices.push(new p5.Vector(_x, _y, 0));
+        this.uvs.push([_x + 0.5, _y + 0.5]);
+
+        this.faces.push([0, (i + 1) % this.detailX + 1, i + 1]);
       }
     };
-    var ellipseGeom = new p5.Geometry(detailX, detailY, _ellipse);
-    ellipseGeom.computeFaces().computeNormals();
-    if (detailX <= 24 && detailY <= 16) {
+    var ellipseGeom = new p5.Geometry(detailX, 1, _ellipse);
+    ellipseGeom.computeNormals();
+    if (detailX <= 50) {
       ellipseGeom._makeTriangleEdges();
       this._edgesToVertices(ellipseGeom);
     } else {
-      console.log(
-        'Cannot stroke ellipse with more' + ' than 24 detailX or 16 detailY'
-      );
+      console.log('Cannot stroke ellipse with more than 50 detailX');
     }
 
     this.createBuffers(gId, ellipseGeom);
   }
-  this.drawBuffers(gId);
+
+  var uMVMatrix = this.uMVMatrix;
```

Comment:
as above, a comment explaining the need to manipulate the matrix here & why drawBuffersScaled isn't used would be helpful to future devs